portal-api/api-tests/v1/gef/amendments/amendments.delete.api-test.ts

```diff
@@ -0,0 +1,129 @@
+
+const validAmendmentId = new ObjectId().toString();
+
+const invalidId = 'invalid-id';
+
+describe('/v1/gef/facilities/:facilityId/amendments/:amendmentId', () => {
+  let testUsers: Awaited<ReturnType<typeof testUserCache.initialise>>;
+  let aMaker: TestUser;
+
+  describe('DELETE /v1/gef/facilities/:facilityId/amendments/:amendmentId', () => {
+    beforeEach(() => {
+      jest.resetAllMocks();
+    });
+
+    beforeAll(async () => {
+      testUsers = await testUserCache.initialise(app);
+      aMaker = testUsers().withRole(MAKER).one() as TestUser;
+    });
+
+    afterAll(() => {
+      jest.resetAllMocks();
+      process.env = originalProcessEnv;
+    });
+
+    describe('when FF_PORTAL_FACILITY_AMENDMENTS_ENABLED is disabled', () => {
+      beforeEach(() => {
+        process.env.FF_PORTAL_FACILITY_AMENDMENTS_ENABLED = 'false';
+      });
+
+      it(`should return a ${HttpStatusCode.NotFound} response`, async () => {
+        // Arrange
+        const url = deleteAmendmentUrl({ facilityId: validFacilityId, amendmentId: validAmendmentId });
+
+        // Act
+        const response = await as(aMaker).remove(url);
+
+        // Assert
+        expect(response.status).toEqual(HttpStatusCode.NotFound);
+      });
+    });
+
+    describe('when FF_PORTAL_FACILITY_AMENDMENTS_ENABLED is not set', () => {
+      beforeEach(() => {
+        delete process.env.FF_PORTAL_FACILITY_AMENDMENTS_ENABLED;
+      });
+
+      it(`should return a ${HttpStatusCode.NotFound} response`, async () => {
+        // Arrange
+        const url = deleteAmendmentUrl({ facilityId: validFacilityId, amendmentId: validAmendmentId });
+
+        // Act
+        const response = await as(aMaker).remove(url);
+
+        // Assert
+        expect(response.status).toEqual(HttpStatusCode.NotFound);
+      });
+    });
+
+    describe('when FF_PORTAL_FACILITY_AMENDMENTS_ENABLED is enabled', () => {
+      beforeEach(() => {
+        process.env.FF_PORTAL_FACILITY_AMENDMENTS_ENABLED = 'true';
+      });
+
+      afterAll(() => {
+        jest.resetAllMocks();
+      });
+
+      withClientAuthenticationTests({
+        makeRequestWithoutAuthHeader: () => remove(deleteAmendmentUrl({ facilityId: validFacilityId, amendmentId: validAmendmentId })),
+        makeRequestWithAuthHeader: (authHeader: string) =>
+          remove(deleteAmendmentUrl({ facilityId: validFacilityId, amendmentId: validAmendmentId }), { headers: { Authorization: authHeader } }),
+      });
+
+      withRoleAuthorisationTests({
+        allowedRoles: [MAKER],
+        getUserWithRole: (role: Role) => testUsers().withRole(role).one() as TestUser,
+        makeRequestAsUser: (user: TestUser) => as(user).remove(deleteAmendmentUrl({ facilityId: validFacilityId, amendmentId: validAmendmentId })),
+        successStatusCode: HttpStatusCode.Ok,
+      });
+
+      it(`should return a ${HttpStatusCode.BadRequest} response when the facility id path param is invalid`, async () => {
```

Comment:
Can we have a test here that checks it returned an error when the amendment id path param is invalid as well?